<!DOCTYPE html>
<html lang="en">
<head>
	<meta http-equiv="X-UA-Compatible" content="IE=edge" />
	<title>PathoTracker huilab</title>
	<meta content='width=device-width, initial-scale=1.0, shrink-to-fit=no' name='viewport' />
	<link rel="icon" href="/statics/assets/img/pku.jpg" type="image/x-icon"/>
	
	<!-- Fonts and icons -->
	<script src="/statics/assets/js/plugin/webfont/webfont.min.js"></script>
	<script>
		WebFont.load({
			google: {"families":["Lato:300,400,700,900"]},
			custom: {"families":["Flaticon", "Font Awesome 5 Solid", "Font Awesome 5 Regular", "Font Awesome 5 Brands", "simple-line-icons"], urls: ['/statics/assets/css/fonts.min.css']},
			active: function() {
				sessionStorage.fonts = true;
			}
		});
	</script>

	<!-- CSS Files -->
	<link rel="stylesheet" href="/statics/assets/css/bootstrap.min.css">
	<link rel="stylesheet" href="/statics/assets/css/atlantis.min.css">
</head>
<body>
	<div class="wrapper">
		<div class="main-header">
			<!-- Logo Header -->
			<div class="logo-header" data-background-color="blue">
				
				<!-- 注意下面是点击这个图片跳转到index.html，我没有改这一步 -->
				<!-- href="index.html" ->改为 href="#"  -->
				<a href="#" class="logo">
					<img src="/statics/assets/img/beidarenmin.jpg" alt="navbar brand" class="navbar-brand" width="40" height="40">
					<img src="/statics/assets/img/pku.jpg" alt="navbar brand" class="navbar-brand" width="40" height="40">
				</a>
				<button class="navbar-toggler sidenav-toggler ml-auto" type="button" data-toggle="collapse" data-target="collapse" aria-expanded="false" aria-label="Toggle navigation">
					<span class="navbar-toggler-icon">
						<i class="icon-menu"></i>
					</span>
				</button>
				<button class="topbar-toggler more"><i class="icon-options-vertical"></i></button>
				<div class="nav-toggle">
					<button class="btn btn-toggle toggle-sidebar">
						<i class="icon-menu"></i>
					</button>
				</div>
			</div>
			<!-- End Logo Header -->

			<!-- Navbar Header -->
			<nav class="navbar navbar-header navbar-expand-lg" data-background-color="blue2">
				
				<div class="container-fluid">
					<span class="text-white fw-bold col-xl-4 ml-auto mr-auto" style="font-size:20px">PathoTracker</span>
				</div>
			</nav>
			<!-- End Navbar -->
		</div>

		<!-- Sidebar -->
		<div class="sidebar sidebar-style-2">			
			<div class="sidebar-wrapper scrollbar scrollbar-inner">
				<div class="sidebar-content">
					<ul class="nav nav-primary">
						<li class="nav-section">
							<span class="sidebar-mini-icon">
								<i class="fa fa-ellipsis-h"></i>
							</span>
							<h4 class="text-section">Components</h4>
						</li>
						<li class="nav-item">
							<a href="/index/">
								<i class="fas fa-desktop"></i>
								<p>Home</p>
							</a>
						</li>
						<li class="nav-item">
							<a href="/Tutorial/"> 
								<i class="fas fa-th-list"></i>
								<p>Tutorial</p>
							</a>
						</li>
						<li class="nav-item">
							<a href="/PathoTracker/">
								<i class="fas fa-pen-square"></i>
								<p>PathoTracker</p>
							</a>
						</li>

						<li class="nav-item">
							<a href="/PathoTracker_ST11/">
								<i class="fas fa-pen-square"></i>
								<p>PathoTracker (ST11)</p>
							</a>
							
						</li>

						<li class="nav-item">
							<a href="/species/">
								<i class="fas fa-chart-bar"></i>
								<p>Species detection</p>
							</a>
						</li>

						<li class="nav-item">
							<a href="/phylogenetic_tree/">
								<i class="fas fa-tree"></i>
								<p>phylogenetic tree</p>
							</a>
						</li>

						<li class="nav-item">
							<a href="/Document/">
								<i class="fas fa-table"></i>
								<p>Document</p>
							</a>
						</li>

						
						<li class="nav-item">
							<a href="/About_Contact/">
								<i class="fab fa-telegram"></i>
								<p>About & Contact</p>
							</a>
						</li>
						
						<li class="nav-item">
							<a href="/Disclaimer/">
								<i class="fas fa-bars"></i>
								<p>Disclaimer</p>
							</a>
						</li>
					</ul>
				</div>
			</div>
		</div>
		<!-- End Sidebar -->

		
		<div class="main-panel">
			<div class="content">
				<div class="page-inner">
					<div class="page-header">
						<h4 class="page-title">Document</h4>
						<ul class="breadcrumbs">
							<li class="nav-home">
								<a href="#">
									<i class="flaticon-home"></i>
								</a>
							</li>
							<li class="separator">
								<i class="flaticon-right-arrow"></i>
							</li>
							<li class="nav-item">
								<a href="#">Home</a>
							</li>
							<li class="separator">
								<i class="flaticon-right-arrow"></i>
							</li>
							<li class="nav-item">
								<a href="#">Document</a>
							</li>
						</ul>
					</div>
					<div class="row">
						<div class="col">
							<div class="card">
								<div class="card-header">
									<h4 class="card-title">Welcome to join our CRE network!</h4>
								</div>
								<div class="card-body">
									<p>
										If you want to join us, please feel free to contact us here.<br>
										We will contact you by email according to the basic form below.
									</p>
									<div class="form-group">
										<label for="affiliation">Affiliation</label>
										<input type="text" class="form-control" id="affiliation" placeholder="Enter affiliation" style="width:60%">
									</div>

									<div class="form-group">
										<label for="institution">Institution</label>
										<input type="text" class="form-control" id="institution" placeholder="Enter institution" style="width:60%">
									</div>

									<div class="form-group">
										<label for="name">Name</label>
										<input type="text" class="form-control" id="name" placeholder="Enter your name" style="width:60%">
									</div>

									<div class="form-group">
										<label for="country">Country</label>
										<input type="text" class="form-control" id="country" placeholder="Enter country" style="width:60%">
									</div>

									<div class="form-group">
										<label for="city">City</label>
										<input type="text" class="form-control" id="city" placeholder="Enter city" style="width:60%">
									</div>
									<div class="form-group">
										<button class="btn btn-success" onclick="sendemail();">Send email</button>
									</div>
									<div class="form-group">
										<small id="emailHelp" class="form-text text-muted">
											This button will automatically email to the manager,
											we will give you feedback in a timely manner after receiving the email.<br>
											Thank you for your participation!<br>
										</small>
									</div>
								</div>
							</div>

							<div class="card">
								<div class="card-header">
									<h4 class="card-title">Database Introduction Documentation</h4>
								</div>
								<div class="card-body">
									<ul class="nav nav-pills nav-secondary" id="pills-tab1" role="tablist">
										<li class="nav-item">
											<a class="nav-link active" id="cre_intro-tab" data-toggle="pill" href="#cre_intro" role="tab" aria-controls="cre_intro" aria-selected="true">CRE Network</a>
										</li>
										<li class="nav-item">
											<a class="nav-link" id="pubdb_intro-tab" data-toggle="pill" href="#pubdb_intro" role="tab" aria-controls="pubdb_intro" aria-selected="false">Isolate form public datasets</a>
										</li>
										<li class="nav-item">
											<a class="nav-link" id="vir_res_db_intro-tab" data-toggle="pill" href="#vir_res_db_intro" role="tab" aria-controls="vir_res_db_intro" aria-selected="false">Virulence and antimicrobial genes database</a>
										</li>
									</ul>
									<div class="tab-content mt-2 mb-3" id="pills-tabContent1">
										<div class="tab-pane fade show active" id="cre_intro" role="tabpanel" aria-labelledby="cre_intro-tab">
											<p>Our research group established a Carbapenem-resistant <i>Enterobacteriaceae</i> (CRE) network to investigate the epidemiology of CRE in China starting from 2014.<br>
												There were 2 stages to this study: first, from 2012 to 2013, we collected 150 CRE isolates from 16 tertiary hospitals, and second, starting in 2014, we expanded the collection area. <br>
												The number of participating hospitals increased from 25 in 2014 to 65 in 2016. <br>
												Peking University People’s Hospital was the lead unit in this project and was responsible for the collection, identification, and sorting of isolates.</p>
											<p></p>
										</div>
										<div class="tab-pane fade" id="pubdb_intro" role="tabpanel" aria-labelledby="pubdb_intro-tab">
											<p>We downloaded 846 <i>Klebsiella pneumoniae</i> isolates form refseq database, did the kleborate type, and incorporated them into this database</p>
											<p></p>
										</div>
										<div class="tab-pane fade" id="vir_res_db_intro" role="tabpanel" aria-labelledby="vir_res_db_intro-tab">
											<p>
												After comparing with the strains in this database, we will output the virulence genes and antimicrobial genes carried by
												the strains closest to the isolate. However, since the isolate itself may carry different virulence or antimicrobial genes
												from the strains in the database, for example, a antimicrobial plasmid is obtained.
											</p>
											<p>
												For the sake of insurance, this
												database will also compare the submitted data with the VFDB database and the resfinder database, and output the virulence
												genes and antimicrobial genes of the isolate sequence for comparison.<br>
												If it is a nosocomial outbreak, the evolution of the strain
												can be briefly viewed from this.
											</p>
										</div>
									</div>
								</div>
							</div>

							<div class="card">
								<div class="card-header">
									<h4 class="card-title">Webserver Introduction Documentation</h4>
								</div>
								<div class="card-body">
									<ul class="nav nav-pills nav-secondary" id="pills-tab2" role="tablist">
										<li class="nav-item">
											<a class="nav-link active" id="PathoTracker_principle-tab" data-toggle="pill" href="#PathoTracker_principle" role="tab" aria-controls="PathoTracker_principle" aria-selected="true">PathoTracker</a>
										</li>
										<li class="nav-item">
											<a class="nav-link" id="species_detection_principle-tab" data-toggle="pill" href="#species_detection_principle" role="tab" aria-controls="species_detection_principle" aria-selected="false">Species detection</a>
										</li>
										<li class="nav-item">
											<a class="nav-link" id="phy_tree_principle-tab" data-toggle="pill" href="#phy_tree_principle" role="tab" aria-controls="phy_tree_principle" aria-selected="false">Phylogenetic analysis</a>
										</li>

									</ul>
									<div class="tab-content mt-2 mb-3" id="pills-tabContent2">
										<div class="tab-pane fade show active" id="PathoTracker_principle" role="tabpanel" aria-labelledby="PathoTracker_principle-tab">
											<!--<p>
												<b><span style="text-decoration:underline;color: #090;">For mNGS raw data (fastq), we provide and first remove its human origin and species, and extract Klebsiella pneumoniae sequence<br>
												This will be displayed on the results page</span></b>
											</p>-->

											<p>The purpose of PathoTracker is to solve how to find the isolate closest to the newly sequencing data
												and timely detect the outbreak of the isolates in the large database.<br></p>
											<p>If the sequencing samples to be detected in the same area over a period of time, by comparing with the
												PathoTracker, the results of the output neighboring isolates are similar (here, the closest isolates
												are basically the same with sample), the output virulence genes and antibiotic resistance genes are
												basically the same, and it can be considered that there may have an isolate outbreak.<br></p>
											<p>At the same time, this service can also trace the source of samples and the geographical and phylogenetic
												evolution. In particular, PathoTracker incorporates not only isolates from China but also isolates submitted
												worldwide in public databases, at which point its possible to trace the evolution of isolates in both
												spatial as well as temporal dimensions.<br></p>
											<p>PathoTracker relies on CRE network. For all the self-collected isolates, we measured the minimum inhibitory
												concentration (MIC) of the <b><span style="text-decoration:underline;color: #f00;">23</span></b> antimicrobials by utilizing broth microdilution method.<br><br>
												The <b><span style="text-decoration:underline;color: #f00;">23</span></b> antimicrobials were as follows:<br>
												&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;meropenem (MEM), imipenem (IMP), ertapenem (ETP), cefoxitin (FOX), cefepime (FEP),<br>
												&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ceftazidime (CAZ) ,ceftazidime/avibactam (CAZ‑AVI), ceftazidime/clavulanic acid (CLC), cefotaxime (CTX),<br>
												&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;ceftriaxone (CRO), Cefoperazone/sulbactam (CSL), Piperacillin/tazobactam (TZP), amikacin (AMK), ciprofloxacin (CIP),<br>
												&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;levofloxacin (LVX), meromycin (MIN), colistin (COL), tigecycline (TGC), chloramphenicol (CHL),<br>
												&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;fosfomycin (FOS), aztreonam (ATM), aztreonam/avibactam (ATM-AVI), cefotaxime/clavulanic acid (CTC)<br>
											</p>
											<p>The greatest advantage of this service is the ability to report genomic and susceptibility information for isolates close
												to the sample. Such that the clinic is capable for dramatically reducing the time required for pure culture, give patients
												timely precise medication, symptomatic treatment, rescue patients at risk, and restrain abuse of antibiotics, based on
												genomic information and self-collected susceptibility information from PathoTracker service report.</p>
										</div>
										<div class="tab-pane fade" id="species_detection_principle" role="tabpanel" aria-labelledby="species_detection_principle-tab">
											<!--<p>Isolate sequences will be cleaned by filtlong</p>
											<p>minimap2 will be used to remove human reads</p>-->
											<p>Centrifuge will be used to detect species</p>
											<br>
										</div>
										<div class="tab-pane fade" id="phy_tree_principle" role="tabpanel" aria-labelledby="phy_tree_principle-tab">
											<p>Isolate sequences will be annotated using Prokka</p>
											<p>Roary will be used to calculate the core genome for phylogenetic analysis, which takes annotated assemblies in GFF3 format (produced by Prokka)</p>
											<p>Iqtree2 will be used to infer phylogenetic trees by maximum likelihood in core genome alignment</p>
											<br>
											<p>Since the isolate genome is large magnitude, so it takes a long time to obtain the genome sequence for
												constructing the phylogenetic tree by conventional mapping and find SNP or MEGA,
												PathoTracker uses Roary to extract the core genome.</p>
										</div>

									</div>
								</div>
							</div>
						</div>

					</div>
				</div>
			</div>
			<footer class="footer">
				<div class="container-fluid">
					<nav class="pull-left">
						<ul class="nav">
							<li class="nav-item">
								<a class="nav-link" href="https://www.pkuph.cn/">
									pkuph
								</a>
							</li>
							<li class="nav-item">
								<a class="nav-link" href="https://www.pku.edu.cn/">
									pku
								</a>
							</li>
							<li class="nav-item">
								<a class="nav-link" href="https://www.bjmu.edu.cn/">
									bjmu
								</a>
							</li>
							<li class="nav-item">
								<a class="nav-link" href="https://www.wanghuilab.net">
									huilab
								</a>
							</li>
						</ul>
					</nav>
					<div class="copyright ml-auto">
						Copyright 2023 Huilab, Department of Clinical Laboratory<br>Peking University People's Hospital, Beijing, China.
						<br>
						<!--<i class="fa fa-heart heart text-danger"></i>
						by <a href="https://www.wanghuilab.net"> Huilab </a>-->
					</div>				
				</div>
			</footer>
		</div>
		
	</div>
	
	<!--   Core JS Files   -->
	<script src="/statics/assets/js/core/jquery.3.2.1.min.js"></script>
	<script src="/statics/assets/js/core/popper.min.js"></script>
	<script src="/statics/assets/js/core/bootstrap.min.js"></script>
	<!-- jQuery UI -->
	<script src="/statics/assets/js/plugin/jquery-ui-1.12.1.custom/jquery-ui.min.js"></script>
	<script src="/statics/assets/js/plugin/jquery-ui-touch-punch/jquery.ui.touch-punch.min.js"></script>
	
	<!-- jQuery Scrollbar -->
	<script src="/statics/assets/js/plugin/jquery-scrollbar/jquery.scrollbar.min.js"></script>
	<!-- Atlantis JS -->
	<script src="/statics/assets/js/atlantis.min.js"></script>

	<script type="text/javascript" >

		function sendemail() {
			var xhr = new XMLHttpRequest();

			var form = new FormData();

			var affiliation_value = $("#affiliation").val()
			form.append('affiliation', affiliation_value);
			console.log(affiliation_value)

			var institution_value = $("#institution").val()
			form.append('institution', institution_value);
			console.log(institution_value)

			var name_value = $("#name").val()
			form.append('name', name_value);
			console.log(name_value)

			var country_value = $("#country").val()
			form.append('country', country_value);
			console.log(country_value)

			var city_value = $("#city").val()
			form.append('city', city_value);
			console.log(city_value)

			//发送表单

			xhr.upload.addEventListener('progress',on_progress,false);
			xhr.open('POST','/join_cre/',true);
            xhr.setRequestHeader('X-CSRFTOKEN','{{ request.COOKIES.csrftoken }}');   //Django CSRF
			xhr.send(form);   //发送表单

			console.log("READYSTATE "+ xhr.status, xhr.readyState)

            xhr.onreadystatechange = () => {
            if ((xhr.status >= 200 && xhr.status < 300) || xhr.status === 304)
            {
                if (xhr.response == 'success'){
                    console.log('in this if');
		            swal("Success!", "Success send email!", "Success send email!")
	            }
				else
				{
					swal("Error!", "Fail to send email!", "Something wrong!")
				}
            }
            else
            {
                console.log("wrong condition");
            }
        };

		}

	</script>
</body>
</html>